### ISSUE TITLE
add a helper function to set all Date to IDate in place

### ISSUE THREAD

**Issue by Sage0614** [Rdatatable/data.table#5646]
I use parquet format to store most of my cached data and use the arrow package to read/write parquet file into data.table, the issue is that the IDate cols are always converted to Date when I read from a parquet stored file. This is not an issue on the data.table package end, but I wonder if it make sense to have a helper function to convert all 'Date' class cols to IDate bundled with data.table, as parquet is such a popular storage format. 

what I want to achieve is something like:

arrow::read_parquet(<file_path_to_a_data.table_stored_in_parquet_format>) %>% data.table::setIDate()

obviously I can have that as a local helper function myself, but I think it would be a nice to have thing for many people.

**Comment by ben-schwen** [comment]
I don't know what `read_parquet` exactly returns, but in any case you should not use any `data.table::set` functions on non `data.table`s.

Since the `Date` to `IDate` conversion is pretty fast (since it is only a type coercing from integer to double), I think that's more of a case for a user-defined function.

**Comment by MichaelChirico** [comment]
I agree. Perhaps the {arrow} folks would be interested in adding support for `read_parquet()` to return integer-backed dates.

**Comment by jangorecki** [comment]
Column has to be copied/allocated anyway, int is 4 bit and double 8bit. So in-place won't really work here.
